Terraform Docs (#3174)
hashicorp/web-unified-docs: content/terraform-policy/v0.2.x (beta)/docs/policy/reference/cli/test.mdx

---
page_title: '`tfpolicy test` command reference'
description: >-
  Reference for the tfpolicy test command to run policy tests locally.
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-07-14T11:44:47-05:00
last_modified: 2026-07-14T19:01:21.000Z
# END AUTO GENERATED METADATA
---

# `tfpolicy test`

The `tfpolicy test` command runs policy tests to verify that your policies work as expected.

@include "beta.mdx"

## Usage

```shell-session
$ tfpolicy test [options]
```

## Description

Once you have written your policy tests, run them using the `tfpolicy test` command. You can provide the path to both the policy files and the test files. If you only specify the path to your policy files, the tests in that path will be executed.

> if you specify targets in your test files with the [`targets`](/terraform/policy/reference/test/policytest#targets) block, Terraform policy will error if a target does not match at least one policy.

Author: ```suggestion
if you specify target policy files or directories in your test files with the [`targets`](/terraform/policy/reference/test/policytest#targets) block, Terraform policy will error if a target does not match at least one policy file.
```

> if you specify target policy files or directories in your test files with the [`targets`](/terraform/policy/reference/test/policytest#targets) block, Terraform policy will error if a target does not match at least one policy file.

Reviewer: ```suggestion
You can target specific policy files or directories in your test files by adding a [`targets`](/terraform/policy/reference/test/policytest#targets) block to your test configuration. Terraform policy reports an error if a target does not match at least one policy file.
```

## Available options

The `tfpolicy test` command supports the following options.

| Option | Description | Default | Required |
|--------|-------------|---------|----------|
| `--policies=<path>` | Test the policies found in the specified path. | Current directory | No |
| `--tests=<path>` | Run the tests found in the specified path. | Same as `--policies` | No |
| `--evaluation-stage=<stage>` | Test policies for the specified evaluation stage: `"setup"` or `"plan"`. | `"plan"` | No |
| `--input="<key>=<value>"` | Set an input value. Can be repeated for multiple inputs. | None | No |
| `--input-file=<path>` | Set input values from a file. The file format is the same as Terraform [.tfvars](/terraform/language/values/variables#variable-definition-files) files. | None | No |

## Exit code

> The `tfpolicy test` command will exit with a non-zero exit code when any tests fail, or an exit code of `0` if all tests pass.

Reviewer: ```suggestion
The `tfpolicy test` command exits with either a non-zero exit code when any tests fail or an exit code of `0` if all tests pass.
```

Once you have written your policy tests, run them using the `tfpolicy test` command. You can provide the path to both the policy files and the test files. If you only specify the path to your policy files, the tests in that path will be executed.

> if you specify targets in your test files with the [`targets`](/terraform/policy/reference/test/policytest#targets) block, Terraform policy will error if a target does not match at least one policy.

Reviewer 2: Could we say "specify target .policy.hcl files" so the user knows what a target means without going to the reference?

"Terraform policy will error if a target does not match at least one .policy.hcl file"

cc: @viveksinghgit95 for second review.

Reviewer 3: Agreed, though a target can be a `.policy.hcl` file or a directory containing one.
